runbotics/runbotics-scheduler/src/auth/microsoft.session.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable, Scope } from '@nestjs/common';
+import { Logger } from '../utils/logger';
+import Axios from 'axios';
+export interface AuthData {
+    token: string;
+    refreshToken: string;
+    expires: number;
+}
+
+interface APIToken {
+    access_token: string;
+    refresh_token: string;
+    expires_on: number;
+}
+
+@Injectable({ scope: Scope.DEFAULT })
+export class MicrosoftSession {
+    constructor() { }
+    private readonly logger = new Logger(MicrosoftSession.name);
+    private session: AuthData = {
+        token: undefined,
+        refreshToken: undefined,
+        expires: undefined,
+    };
+
+    private readonly TENANT_ID = process.env.SHAREPOINT_TENANT_ID;
+    private readonly CLIENT_ID = process.env.SHAREPOINT_CLIENT_ID;
+    private readonly USERNAME = process.env.SHAREPOINT_USERNAME;
+    private readonly PASSWORD = process.env.SHAREPOINT_PASSWORD;
+    private readonly RESOURCE = 'https://graph.microsoft.com';
+    private readonly CALLBACK_URL = 'http://localhost:3000';
+    private readonly GRANT_TYPE = 'password';
+    private readonly TOKEN_URL = `https://login.microsoftonline.com/${this.TENANT_ID}/oauth2/token`;
+
+    private readonly params = {
+        client_id: this.CLIENT_ID,
+        grant_type: this.GRANT_TYPE,
+        resource: this.RESOURCE,
+        username: this.USERNAME,
+        password: this.PASSWORD,
+        callbackURL: this.CALLBACK_URL,
+    };
+
+    public async getToken() {
+        if (this.session.token) {
+            this.logger.log(`Using existing token...`);
+            return this.session;
+        } else {
+            this.logger.log(`Getting new token...`);
+            return MicrosoftSession.authorize(this.TOKEN_URL, this.params);
+        }
+    }
+
+    public static async authorize(url: string, params: { [key: string]: string }) {
+        const CancelToken = Axios.CancelToken.source();
```

Comment:
our variables should be in camel case, do not bother with axios naming convention